### ISSUE TITLE
mat-stepper Not possible to make a step un-editable if it's after the current one

### ISSUE THREAD

**Issue by NothingEverHappens** [angular/material2#14719]
#### What is the expected behavior?
If mat-step has `[editable]="false"`, it's not possible to select it.

#### What is the current behavior?
It's possible to select a step with `[editable]="false"`, if it's after the current step

#### What are the steps to reproduce?
https://stackblitz.com/edit/angular-material2-issue?file=app%2Fapp.component.html

Click on step 3, it's possible to jump to it despite it not being editable.
It's not possible to go back to step 1

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Latest

**Comment by crisbeto** [comment]
@NothingEverHappens it seems like the Stackblitz link is pointing to the Material starter. You may have forgotten to save your changes.

**Comment by NothingEverHappens** [comment]
Thanks, updated! 
https://stackblitz.com/edit/angular-material2-issue-crupyb?file=app/app.component.html

**Comment by Ancyker** [comment]
> Looking at how the logic is set up now, it seems like this is the intended behavior. Also see the description for the `editable` property:
> 
> > Whether the user can return to this step once it has been marked as completed.

Not really, you can bypass it entirely by going back to a step before a completed step and then progressing forward. If completed is true and editable is true, the step should be skipped.

**Comment by Pajchel** [comment]
@Ancyker did you find solution for this issue? I have exactly the same problem :(
Thanks

**Comment by PMorais13** [comment]
I have the exact same problem, if anyone knows please share

**Comment by jelbourn** [comment]
Duplicate of #17294
